### PR TITLE
#3566 Success message can appear a couple of times

### PR DESCRIPTION
**Pull request by VanilStorm**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/3566

**Problem:**
* Success message can appear a couple of times if user often clicks on wishlist sign on PDP

**In this PR:**
* Created a check whether there is a product in the wish list removed the possibility of several clicks in a row

### CODE REVIEW

**Comment by Elena-Evsa** [comment]
1. Endless loading appears after some clicks on wishlist sign https://vimeo.com/691270988/6fbe2b1c1e

**Comment by VanilStorm** [comment]
Try this link ( https://pr-4491.d25kd40decjpue.amplifyapp.com/default/8bunfixed.html )
Same problem ?

**Comment by VanilStorm** [comment]
Strange, everything is fine on my screen
Okay, I'll try to do something.

**Comment by Elena-Evsa** [comment]
Can reproduce with ( https://pr-4491.d25kd40decjpue.amplifyapp.com/default/8bunfixed.html )

**Comment by VanilStorm** [comment]
Something is wrong with the production ( https://scandipwapmrev.indvp.com/default/grouped-e-reader.html) , I can't test my code. 
There are some problems with the loader and adding to the wish list, an error is displayed when trying to add it or endless loading animation.

**Comment by VanilStorm** [comment]
And when you are in the PDP and refresh the page, it is mirrored
Снимок экрана 2022-03-23 в 14 13 31

**Comment by VanilStorm** [comment]
but for some reason this does not always happen

File: packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js [hunk 1]
```diff
@@ -108,23 +109,35 @@ export class ProductWishlistButtonContainer extends PureComponent {
...
+        } else {
+            return;
+        }
 
         if (add) {
             await addProductToWishlist({
                 items: magentoProduct,
                 wishlistId
-            });
+            }).then(() => this.setState({ isWishListToggle: false }));
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js:132]: Why then? You anyway awaiting for addProductToWishlist, if you will put state setting after it I guess result will be same

File: packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js [hunk 2]
```diff
@@ -108,23 +109,35 @@ export class ProductWishlistButtonContainer extends PureComponent {
...
 
             return;
         }
 
         const wishlistItem = this.getWishlistItem(sku);
+
         if (!wishlistItem) {
+            this.setState({ isWishListToggle: false });
             return;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js:141]: Space before return

File: packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js [hunk 3]
```diff
@@ -134,6 +147,7 @@ export class ProductWishlistButtonContainer extends PureComponent {
             }
         } = wishlistItem;
 
+        this.setState({ isWishListToggle: false });
         return removeProductFromWishlist({ item_id: itemId });
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js:151]: space before return

File: packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js [hunk 4]
```diff
@@ -108,23 +109,35 @@ export class ProductWishlistButtonContainer extends PureComponent {
...
             return showNotification('info', __('You must login or register to add items to your wishlist.'));
         }
 
         this.setWishlistButtonLoading(true);
+        
+        if (!isWishListToggle) {
+            this.setState({ isWishListToggle: true });
+        } else {
+            return;
+        }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js:126]: make it simpler. 
```JS
        if (isWishListToggle) {
            return;
        }
        
        this.setState({ isWishListToggle: true });
```

File: packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js [hunk 5]
```diff
@@ -98,7 +99,7 @@ export class ProductWishlistButtonContainer extends PureComponent {
         return this.setState({ isWishlistButtonLoading: isLoading });
     }
 
-    async toggleProductInWishlist(add = true) {
+    async toggleProductInWishlist(add) {
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js:102]: There is a reason for it?

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js (hunk 1 above)
Comment (VanilStorm) [reply packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js:132]: I understand this, I didn't notice it when refactor code

File: packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js (hunk 5 above)
Comment (VanilStorm) [reply packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js:102]: In my opinion, this is not always safe, if somehow the value does not come to the function, then another error may pop up. Maybe you need to add a check for the value, whether it exists or not. Or maybe I'm wrong)

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Pull request merged**